scripts/upgrade-via-npm.ps1

```
@@ -96,365 +166,801 @@ if (-not $CMD) {
+function Invoke-NpmPack {
+    $script:PACK_TMP_DIR = Join-Path ([System.IO.Path]::GetTempPath()) "qqbot-pack-$([guid]::NewGuid().ToString('N').Substring(0,8))"
+    New-Item -ItemType Directory -Path $script:PACK_TMP_DIR -Force | Out-Null
+    $script:PACK_TGZ_FILE = ""
+
+    $ok = $false
+    foreach ($registry in $NPM_REGISTRIES) {
+        $packDest = $script:PACK_TMP_DIR
+        $packSrc = $INSTALL_SRC
+        $packRegistry = $registry
+
+        $result = Run-WithTimeout -TimeoutSecs $Timeout -Description "npm pack" -ScriptBlock {
```

PSScriptAnalyzer (1.25.0)

[warning] The variable 'result' is assigned but never used.

(PSUseDeclaredVarsMoreThanAssignments)